torch/csrc/jit/codegen/cuda/predicate_compute.cpp

```diff
@@ -155,33 +164,28 @@ kir::Bool* PredicateCompute::getInlinePredicate(
 
   std::vector<kir::Bool*> preds;
 
-  for (auto pred : all_preds)
-    if (!(pred->isConst()) || !(pred->isConst() && pred->value().value()))
+  for (auto pred : all_preds) {
+    if (!pred->isConst() || !(pred->isConst() && pred->value().value())) {
       preds.push_back(pred);
+    }
+  }
 
   if (preds.empty()) {
     return ir_builder.create<kir::Bool>(true);
   }
 
-  Val* cond = preds[0];
-
-  for (decltype(preds.size()) i{1}; i < preds.size(); i++) {
+  kir::Val* cond = preds[0];
+  for (size_t i = 1; i < preds.size(); i++) {
     cond = ir_builder.andExpr(cond, preds[i]);
   }
 
-  TORCH_INTERNAL_ASSERT(
-      cond->getValType().value() == ValType::KirScalar &&
-          cond->getDataType().value() == DataType::Bool,
-      "Error computing predicate, should be returning a Bool, but returning ",
-      cond->getDataType().value());
-
```

Comment:
Why removing this check?

Reply:
It was just checking that the andExpr() returns a bool, which is below the bar even for paranoid checks (we don't check the type of every expression, are we?)

Reply:
That's true.